csrc/dynamic_transform.cpp

```diff
@@ -370,18 +447,34 @@ class DynamicTransformConcretizer : public OptOutMutator {
 };
 
 void DynamicTransformConcretizer::concretize() {
-  // First, concretize all dynamic reshape ops
+  // Concretize all dynamic reshape ops
   concretizeReshape();
 
   // Set output IterTypes for dynamic resize ops
   concretizeResize();
 
+  // Registers replacement of all empty extents with zeroVal()
+  concretizeEmptyExtents();
+
   // Finally, propagate concretized domains
-  auto all_stmts = StmtSort::getStmts(info_->fusion(), true);
-  for (auto stmt : all_stmts) {
-    if (stmt->isA<Val>()) {
-      mutate(stmt);
+  auto all_stmts = StmtSort::getStmts(info_->fusion());
```

Comment:
Note we no longer traverse into members.

Reply:
Don't remember why it did

Reply:
I changed it to `true` in #258 so that the traversal would handle `IterDomain`s, but I didn't have a clear enough picture of how that should work at that time.

Reply:
Now we explicitly are only traversing the `TensorView` graph, and we only handle `IterDomain` members manually.